src/mpyl/utilities/repo/__init__.py

```diff
@@ -116,6 +132,37 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             raise exc_val
         return self
 
+    @staticmethod
+    def from_clone(config: RepoConfig, repo_path: Path):
+        creds = config.repo_credentials
+        if not creds:
+            raise ValueError("Cannot clone repository without credentials")
+
+        repo = Repo.clone_from(
+            url=creds.to_url_with_credentials,
```

Comment:
Actually I have one more question, why did you change this to use the username/password way instead of just the ssh url? It even warns that the support for this has been dropped on August 13, 2021

Reply:
Ah yes. The reason I used the ssh URL is that it was a way to get it to work in github actions: https://github.com/SamTheisens/mpyl-example-gha/pull/6/files#diff-ee8dae421069ba04c12f6746ff297be2674d2dbd629e36c374e5f07245fdec71R11-R14
I don't have ssh credentials there. 

But if you need ssh access on Jenkins, I think you need to add that feature. I can't really test that myself because I don't have access to Jenkins. Would be good if we keep the possibility to use either.